new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,438 @@ description: Learn about writing Oso policies - the source of truth for authoriz
+most applications use more structured representations for their
+actors, actions, and resources.
 
-role(user, role_name) if
-    user.role = role_name;
-```
+Suppose then that our actors are represented by instances of a
+{{< lang >}} `User` class, with, say, a user ID and administrator
+flag:
 
-`role(user, "accountant")` is another example of pattern matching in Polar. Any
-time a rule body contains a **predicate** like this, it is performing another
-query. That is, it will try to find all *matching* rules named `role` with two
-inputs.
+```python
+@dataclass
+class User:
+    id: int
+    admin: bool
+```
 
-You can also either use the [REPL](repl) or the `Oso.query_rule()` method to
-interact with this directly. For example:
+Let's also assume also a simple `Resource` class, with an `owner`
+field that references a user ID:
 
 ```python
-from oso import Oso
+@dataclass
+class Resource:
+    owner: int
+```
 
-class User:
-    def __init__(self, name, role):
-        self.name = name
-        self.role = role
+The two policy rules we'll implement are:
+
+* A user that is an administrator may read any resource.
+* A user may read any resource that they own.
 
-oso = Oso()
-oso.load_str("role(user, role_name) if user.role = role_name;")
+Our application will be making calls like this:
 
-alice = User("alice", "accountant")
-assert list(oso.query_rule("role", alice, "accountant"))
+```python
+oso.is_allowed(User(id=0, admin=True), "read", Resource(owner=0))
 ```
 
-{{% callout "Tip" "green" %}}
-  Try setting the `POLAR_LOG` environment variable before executing a polar
-  query to see a [trace](tracing) of how the query is evaluated:
+We can implement our policy rules by binding `user` and `resource`
+variables to the first and third arguments, then using the `.`
+("dot") operator to lookup field values in the bodies:
 
-  ```console
-  $ POLAR_LOG=1 python user.py
-  [debug]   QUERY: role(<__main__.User object at 0x105da6190>, "accountant"), BINDINGS: {}
-  [debug]     APPLICABLE_RULES:
-  [debug]       role(user, role_name) if user.role = role_name;
-  [debug]     RULE: role(user, role_name) if user.role = role_name;
-  [debug]       QUERY: .(_user_5, "role", _value_1_7) and _value_1_7 = _role_name_6, BINDINGS: {_role_name_6 = "accountant", _user_5 = <__main__.User object at 0x105da6190>}
-  [debug]         QUERY: .(_user_5, "role", _value_1_7), BINDINGS: {_user_5 = <__main__.User object at 0x105da6190>}
-  [debug]           LOOKUP: <__main__.User object at 0x105da6190>.role()
-  [debug]           => "accountant"
-  [debug]         QUERY: _value_1_7 = _role_name_6, BINDINGS: {_role_name_6 = "accountant", _value_1_7 = "accountant"}
-  [debug]   BACKTRACK
-  [debug]           LOOKUP: <__main__.User object at 0x105da6190>.role()
-  [debug]           => No more results.
-  [debug]           BACKTRACK
-  [debug]           HALT
-  ```
-{{% /callout %}}
+```polar
+allow(user, "read", resource) if
+    user.admin = true;
```

Comment:
```suggestion
```polar
# A user that is an administrator may read any resource. 
allow(user, "read", resource) if
    user.admin = true;
```